iOS/MMIWG2S/ViewControllers/FacesViewController.swift

```diff
@@ -206,25 +221,42 @@ public final class FacesViewController: UIViewController {
 extension FacesViewController: ARFaceViewerUIDelegate {
     
     func textureChanged(toRed: Bool) {
-        // TODO: get these to work without crashing app
         if toRed {
-//            self.faceTextureMaterial.diffuse.contents = UIImage(named: "Face.scnassets/red_hand_full_mouth.png")
+            self.faceTextureMaterial.diffuse.contents = UIImage(named: "Face.scnassets/red_hand_full_mouth.png")
         } else {
-//            self.faceTextureMaterial.diffuse.contents = UIImage(named: "Face.scnassets/black_hand_80_texture.png")
+            self.faceTextureMaterial.diffuse.contents = UIImage(named: "Face.scnassets/black_hand_80_texture.png")
         }
     }
     
     func generateShareImage() -> UIImage? {
+        isCapturing = true
         let deviceOrientation = UIDevice.current.orientation
         guard let pixelBuffer = faceSession?.currentFrame?.capturedImage,
-              let image = UIImage(pixelBuffer: pixelBuffer, orientation: .from(deviceOrientation: deviceOrientation))
-              else { return nil }
+              let capturedImage = UIImage(pixelBuffer: pixelBuffer, orientation: .from(deviceOrientation: deviceOrientation))
+              else {
+            isCapturing = false
+            return nil
+        }
+        
         captureSession?.stopRunning()
-        return image
+        var sceneImage = sceneView.snapshot()
+        
+        if deviceOrientation.isLandscape {
+            sceneImage = sceneImage.rotate(radians: deviceOrientation == .landscapeLeft ? .pi / 2 : -.pi / 2) ?? sceneImage
+        }
+        UIGraphicsBeginImageContextWithOptions(capturedImage.size, true, 0.0)
+        capturedImage.draw(in: CGRect(origin: .zero, size: capturedImage.size))
+        sceneImage.draw(in: CGRect(origin: .zero, size: capturedImage.size))
```

Comment:
How do these gets used? I see they are drawn, just not sure how they are used :)

Reply:
its drawn to a static instance. `UIGraphicsGetImageFromCurrentImageContext` reads from the current context we were just writing to